pkg/operator/controller/dns_status.go

```diff
@@ -31,10 +39,47 @@ func (r *reconciler) syncDNSStatus(dns *operatorv1.DNS, clusterIP, clusterDomain
 	return nil
 }
 
-// computeDNSStatusConditions computes dns status conditions based on
-// the status of ds and clusterIP.
-func computeDNSStatusConditions(oldConditions []operatorv1.OperatorCondition, clusterIP string,
-	ds *appsv1.DaemonSet) []operatorv1.OperatorCondition {
+// checkEndpoints verifies that the provided endpoints by sending a DNS query
+// for the API service host name to each endpoint and returns a list of errors.
+func checkEndpoints(endpoints *corev1.Endpoints) []error {
+	const apiHost = "kubernetes.default.svc.cluster.local."
+	var (
+		errs      []error
+		addresses []string
+	)
+	for _, ep := range endpoints.Subsets {
```

Comment:
How should this function behave if  `endpoints` is empty? Similarly, should this function (or maybe a different function) check that the expected number of endpoints are being evaluated?

Reply:
I added a check that adds an error if the number of addresses does not match the number of available pods as reported by the daemonset.  I think this suffices as `computeDNSDegradedCondition` sets `Degraded=True` if the daemonset reports that no pods are available.